TFT/src/User/Menu/common.c

```diff
@@ -185,21 +185,38 @@ const void drawStandardValue(const GUI_RECT *rect, VALUE_TYPE valType, const voi
 
 const bool warmupTemperature(uint8_t toolIndex, void (* callback)(void))
 {
-  if (heatGetCurrentTemp(toolIndex) < infoSettings.min_ext_temp)
-  { // low temperature warning
+  #define TEMP_OFFSET 5  // offset temperature to avoid denial of extrusion/retraction due to the nozzle temperature lag
+
+  if (heatGetCurrentTemp(toolIndex) < infoSettings.min_ext_temp - TEMP_OFFSET)
+  { // low temperature warning 
     char tempMsg[120];
     LABELCHAR(tempStr, LABEL_EXT_TEMPLOW);
 
     sprintf(tempMsg, tempStr, infoSettings.min_ext_temp);
-    strcat(tempMsg, "\n");
-    sprintf(tempStr, (char *) textSelect(LABEL_HEAT_HOTEND), infoSettings.min_ext_temp);
-    strcat(tempMsg, tempStr);
-
-    setDialogText(LABEL_WARNING, (uint8_t *) tempMsg, LABEL_CONFIRM, LABEL_CANCEL);
-    showDialog(DIALOG_TYPE_ERROR, callback, NULL, NULL);
 
+    if (heatGetTargetTemp(toolIndex) < infoSettings.min_ext_temp)
+    { // heatup offering
+      strcat(tempMsg, "\n");
+      sprintf(tempStr, (char *) textSelect(LABEL_HEAT_HOTEND), infoSettings.min_ext_temp);
+      strcat(tempMsg, tempStr);
+      
+      setDialogText(LABEL_WARNING, (uint8_t *) tempMsg, LABEL_CONFIRM, LABEL_CANCEL);
+      showDialog(DIALOG_TYPE_ERROR, callback, NULL, NULL);
+    }
+    else
+    {
+      setDialogText(LABEL_WARNING, (uint8_t *) tempMsg, LABEL_CONFIRM, LABEL_BACKGROUND);
+      showDialog(DIALOG_TYPE_ERROR, NULL, NULL, NULL);
+    }
```

Comment:
Extrusion should not be blocked when the temperature is above minimum extrusion temperatures, it should be blocked only if the target temperature is set through `M109`.

Reply:
You are right, it should not be blocked. ...and it isnt't blocked, as it should be.

Reply:
warmupTemperature returns false (Line no: 211) regardless of this. which will always prevent extrusion.

Reply:
Yes, I missed line no: 197

Reply:
No problem, thanks for checking.